ledger/src/blockstore_processor.rs

```diff
@@ -444,7 +461,7 @@ fn rebatch_transactions<'a>(
     let results = &lock_results[start..=end];
     let mut tx_batch =
         TransactionBatch::new(results.to_vec(), bank, OwnedOrBorrowed::Borrowed(txs));
-    tx_batch.set_needs_unlock(false);
+    tx_batch.set_needs_unlock(true); // unlock on drop for easier clean up
```

Comment:
nit: no need to explicit call of `.set_needs_unlock` because `TransactionBatch::new` set the flag to `true` automatically. how about like this?

```rust
    // automatically unlock on drop for easier clean up
    let mut tx_batch =
        TransactionBatch::new(results.to_vec(), bank, OwnedOrBorrowed::Borrowed(txs));
```

Reply:
Even though it's redundant, I somewhat like that it's very explicitly showing that the batch is taking over the responsibility of unlocking.
lmk if you still disagree

Reply:
i see. no strong opinion. just wanted to see your opinion.